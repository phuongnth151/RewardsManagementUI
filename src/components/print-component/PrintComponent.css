.print-button-search{
    font-size: 20px;
    color: #FFF;
    text-decoration: solid;
    background-color: #f96302;
    border: 1px solid #f96302;
}

.print-button-disabled{
    font-size: 20px;
    color: #666666;
    text-decoration: solid;
    background-color: #e5e5e5;
    border: 1px solid #666666;
}

.orangeColor{
    color: #f96302;
}
#searchbox:focus{
    outline: none;
}

.btnContainer {
    line-height: 38px;
    width: calc(100% - 2px); 
    font-size: 20px; 
}

.printContainer {
    width: 100%;
    bottom: 0;
    position: absolute;
    text-align: center;
    /* border: 1px solid red; */
    /* background-color: #f96302;     */
}

.printerInfo {
    background-color: #fff !important;
    display: table;
    width: 100%;
    margin: 5% 0;
}

.printerInfo > * {
    display: table-cell;
}

.fontReduced {
    font-size: 13px;
}

.print-button-enabled{
	color: #F96302;
}


