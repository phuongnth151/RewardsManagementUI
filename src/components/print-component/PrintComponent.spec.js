/*eslint-disable */

import React from 'react'
import { shallow, mount } from 'enzyme'
import PrintComponent from './PrintComponent'
import { createMemoryHistory } from 'history'

let component;
let props = {
    location: { pathname: "/" },
};
let historyMock = {   push: jest.fn(),
    location: 
        {pathname: "/" ,state: {fromPage: "test"}} 
};

describe('Test Button component', () => {


    beforeEach(() => {

    });    

    it('Makes sure the <PrintComponent/> will render without any issue', () => {
        const history = createMemoryHistory('/printer')
        let props = {
            location: { pathname: "/" },
        };
        component = shallow(<PrintComponent  {...props}  history={historyMock} />);
    expect(component).toMatchSnapshot();
    });

    it('Makes sure the <PrintComponent/> renders without any issue', () => {
        let props = {
            location: { pathname: "/" },
        };
        component = shallow(<PrintComponent  {...props}  history={historyMock} />)
        expect(component.find('.printContainer').exists()).toBeTruthy();
    })

    it('Test Change Printer Button click event', () => {
        let props = {
            location: { pathname: "/product" },
        }; 
       const history = createMemoryHistory('/printer');
       component = shallow(
            <PrintComponent {...props} history={history} />
            )
        const spyClickHandlerFn = jest.spyOn(
            component.instance(),
            'handleSelectPrinterBtnClick'
        );
        component.find('.change-printer').simulate('click');
        expect(spyClickHandlerFn).toHaveBeenCalledOnce;
    });

    it('Test Print Label Button click event', () => {
        let props = {
            skuNbr: 161640,
            itemContext: {itemQuantity: 1, searchedInput: 161640},
            selectedPrinter: [{
                printerUxQueueName:'rfprs001',
                printerDescription:'TECLBL1'}],
            printLabelClicked: jest.fn(),
            handleLoading: jest.fn(),
            location: { pathname: "/product" }, state: {fromPage: "test"}
        }
        component = mount(<PrintComponent {...props} history={historyMock}/>)

        const spyClickHandlerFn = jest.spyOn(
            component.instance(),
            'handlePrintLabelClicked'
        );

        component.find('.printerInfo').simulate('click');
        expect(spyClickHandlerFn).toHaveBeenCalledOnce;
    });
});