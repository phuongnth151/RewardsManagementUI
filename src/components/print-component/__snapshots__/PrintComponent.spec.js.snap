// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component Makes sure the <PrintComponent/> will render without any issue 1`] = `ShallowWrapper {}`;
